### ISSUE TITLE
Instructions for Baker Shelf

### ISSUE THREAD

**Issue by anjimi** [Simbul/baker#537]
Hello, I see the Baker Shelf is now ready to use, this is great! Thank you for this.

I have a magazine which I have been testing using Baker 3.2, I would now like to test with the Shelf.

I've downloaded the latest shelf files, and following the instructions here: https://github.com/Simbul/baker/wiki/Newsstand-shelf-JSON

I have opened the file Constants.h to set the location of my .json file, I see there are other constants here which need to be defined, PURCHASE_CONFIRMATION_URL and PRODUCT_ID_FREE_SUBSCRIPTION

I don't know how these are used and what I should set them as.

Also please can you confirm, the hpub file is a zip file of the book folder contents, renamed to hpub?

Is there a more full set of instructions somewhere which I have missed?

**Comment by folletto** [comment]
Hello @anjimi, we updated that page a bit and... yes, we are still working on the documentation. :)

Always work in progress, but there are other pages you might look:
- https://github.com/Simbul/baker/wiki/Newsstand-vs-Bundled-publications-support-in-Baker-4.0
- https://github.com/Simbul/baker/wiki/Subscription-pingback

I'm also guessing that you tried the `shelf` branch instead of the `master` branch: `master` is now the only one, and `shelf` contains just an old version. I'm guessing this because in `master` there are already more information in the code. :)

Let us know, it might be helpful to have a better idea on what to write and how to organize the documentation. :)

**Comment by anjimi** [comment]
Thank you, the page on the subscription pingback is useful, and its relevant links to the apple developer site, which I will read up on. I'm now guessing that PRODUCT_ID_FREE_SUBSCRIPTION may relate to the subscription id in iTunes Connect?

I'll download the latest baker master and check that out.

I see now how the baker wiki is organised, sorry I'm still familiarising myself with everything here :)

**Comment by folletto** [comment]
Yes, check the master, it should be clearer there. ;)

Thanks also for saying that the wiki is organized. I feel that "organized" is a bit of a big word, but I'll try to improve it soon. :D

**Comment by bahmet** [comment]
hi,
is there a published magazine with baker 4.0 on appstore, that we can download and see the process.
thanks

**Comment by folletto** [comment]
Hi @bahmet excuse me but I'm not sure I understand you correctly: what do you need exactly? I'm not sure because from a published app you can't see much of the process, and on the other side we are looking for publishers: https://github.com/Simbul/baker/wiki/Launch-test-for-Baker-4.0-(shelf)

**Comment by bahmet** [comment]
hi @folletto no please, you excuse me because i am old fashioned one. i would like to see the startup screen, shelf view, click to download the issue, progress bar etc on an actual device with a publication other than Sherlock Holmes, sorry:)

**Comment by folletto** [comment]
I'd love to see that to. We are looking for publications as I said above hehe. :D

**Comment by ghost** [comment]
@folletto I'm representing a publisher and would like to test the 4.0 version. I was just getting ready to try to publish with the older version like anjimi was. Where would you like me to contact you that we want to be part of the 4.0 launch? I may have many questions and I am more than willing to to help with feedback.

**Comment by folletto** [comment]
Either here (open a new issue for you if you want dedicated open support), or the project's contact email (info AT bakerframework DOT com) are fine. :)

Ideally, by downloading master you should be ready to go. :)

**Comment by bahmet** [comment]
hi @folletto 
i have 2 free applications available for ipad made with baker. one of them is a story book with coloring pages https://itunes.apple.com/us/app/bremen-m-z-kac-lar-interaktif/id507130003?ls=1&mt=8
the other is coloring book https://itunes.apple.com/us/app/egitici-boyama-kitaplar-serisi/id526663518?ls=1&mt=8
i would like to use shelf option to put my apps together. i can be a part of the testing. but i need step by step instructions.
pls let me know if these apps ok to test the shelf.

**Comment by folletto** [comment]
Well, for us it's fine, it's Apple Review the culprit here. :D

The instructions are in the link above: https://github.com/Simbul/baker/wiki/Launch-test-for-Baker-4.0-(shelf)
Could you tell us what's not clear so we can add more details? :)

**Comment by tstegart** [comment]
We will be trying to test this in the next few days, so I'll start a new thread with any issues we have. Thanks for all the hard work everyone, it looks great. I have to say, Apple's documentation on this is a mess, the free subscriptions and paid are all intermingled and the terminology changes from document to document. If I get it working, I'll try write up the process.

**Comment by folletto** [comment]
THANKS. :)

**Comment by folletto** [comment]
We worked a bit on the wiki and now there's a more extensive tutorial here:
https://github.com/Simbul/baker/wiki/4.0-tutorial-for-Newsstand

**Comment by anjimi** [comment]
Our timeframe fits in with the 4.0 release, so may be able to help with testing/approval. Our publication is almost ready. I need to test with latest master baker and look at Apple setup.

New wiki tutorial is great, thank you so much for this. :)

**Comment by folletto** [comment]
That's great news! Thanks. :D

**Comment by anjimi** [comment]
Just found this on Apple Newsstand FAQ: http://developer.apple.com/library/ios/#technotes/tn2280/_index.html

> Is my Newsstand app required to use Newsstand Kit's background downloading feature?
> 
> If you're delivering an issue to a user as part of a subscription, you must deliver it via background download.
> 
> In general, users will expect Newsstand apps to have content downloaded in the background, ready and available for offline viewing. If a user hasn't purchased a subscription for new issues, you may use Newsstand Kit's background downloading feature to deliver other content. Keep in mind that your Newsstand app should not deliver or manage content in a way that ends up consuming large amounts available storage space on a user's device.

This would suggest that push notifications for background download are required for Newsstand apps.

**Comment by Simbul** [comment]
With App Store review policies it's always hard to say something for certain.

However, the way I read that is that you should use the NK download facilities for Newsstand issues - which Baker is already doing.
As far as I can see, NK downloads are separate from push notifications: push notifications are essentially a way of triggering a download.

**Comment by folletto** [comment]
In other words: http://stackoverflow.com/questions/11895035/do-newsstand-apps-require-push-notifications :)

—

Please @anjimi but don't use this issue for discussion, open a new one if you have any doubt around a specific topic, otherwise we can't track, help people and reply to messages properly anymore. :)

**Comment by tstegart** [comment]
Ca I ask what people are using for screenshots? Its a little difficult to take a screenshot of an in-app purchase you can't take a picture of until the in-app purchase has been created. Seems circular.

**Comment by ghost** [comment]
@tstegart if you are able to test the downloaded magazine/book on an iPad through Xcode while plugged into the compter, you can grab screen shots through Xcode.

**Comment by tstegart** [comment]
Thanks. I'm just not sure what to take a picture of. In-app purchases need to be approved by Apple and they want a screenshot. I guess I'll just submit a picture of the Baker shelf. Anyone else got their IAP approved yet? How long did it take?

**Comment by folletto** [comment]
As reported [here on Stackoverflow](http://stackoverflow.com/questions/4802810/iphone-in-app-purchase-screen-shot) it seems that in the _In-App Purchases Guidelines_ there's a paragraph saying:

> For review purposes only. Your In App Purchase screenshot will not display on the App Store or on the iPhone device store. Once you have tested your In App Purchase and are ready to sign off, upload a screenshot of your In App Purchase in this section. The screenshot should be a clear picture of your In App Purchase in action and must be 320 x 480, 480 x 320, 320 x 460, or 480 x 300 pixels.

So... it seems pretty much open. :)

**Comment by mtamony** [comment]
I'm also interested in testing this but am a little confused. There is no real way to test the actual process right? It is all done once the magazine is approved by Apple and in App store?

**Comment by folletto** [comment]
You can test everything before submitting. Just follow Apple instructions on how to setup the sandbox. :)

**Comment by yitweng** [comment]
I'm uploading the Bakershelf's 4.0 latest master Binaries as I speak.... I've got standing version of Andrew's brunch of Bakershelf previously, and have got in-app purchase approve prior. Crossing fingers ;)

btw my app is available for download here :+1: 
http://www.jonlow.com/portfoliobyjonlow

Cheers,

Jon

**Comment by folletto** [comment]
That's great news @yitweng, let us know how it goes, your publications seems very interesting. :)

Did you submit a Standalone ora a Newsstand version of Baker?

**Comment by yitweng** [comment]
Hello,

Yes. It will be the latest baker build with shelf and free sub 

Here is the shelf

https://www.dropbox.com/s/fp9fzmtg4ujhllp/Photo%2015-12-2012%2011%2041%2023%20AM.jpg

Sent via iPhone

**Comment by nin9creative** [comment]
Ha!  Switching shelves?  :)

Andrew

**Comment by yitweng** [comment]
Yeah, we had a good run with the first shelf. Hopefully by taking the existing content up to apple, and if they do approve it, it will be great :)

Sent via iPhone

**Comment by nin9creative** [comment]
Good luck!

Warm Regards,
Andrew

**Comment by folletto** [comment]
Ok, the documentation should be done by now. There's always margin of improvement, but we can close this issue now. :)

Thanks everyone. :)
